src/app/hearing/services/possible-hearing-actions.service.ts

```diff
@@ -48,8 +49,9 @@ export class PossibleHearingActionsService {
         const action: PossibleActionConfig = this.possibleActions[value];
 
         if (action.enabled) {
-            action.openDialog().subscribe((confirmed) => {
-                this.confirmationDialogClosed(confirmed, () => action.callService(hearing), hearing, action.summaryText)
+            action.openDialog().subscribe((confirmationData) => {
```

Comment:
The name 'confirmationData' is misleading becase this variable will have some value even if dialog is not confirmed but declined. Call it 'dialogData' and pass the entire 'dialogData' to the action.callService function. This way you don't bind directly to a particular key (description in this instance) and your mechanism is more generic this way.